tests/integration/test_charm.py

```diff
@@ -43,9 +44,66 @@ async def test_build_and_deploy(ops_test: OpsTest, request: pytest.FixtureReques
 
     assert ops_test.model
     await ops_test.model.deploy(charm, resources=resources, application_name=APP_NAME)
+    await ops_test.model.deploy(
+        "self-signed-certificates",
+        application_name=TLS_PROVIDER_APPLICATION_NAME,
+        channel="edge",
+        trust=True,
+    )
+    await ops_test.model.deploy(
+        "tls-certificates-requirer",
+        application_name=TLS_REQUIRER_APPLICATION_NAME,
+        channel="edge",
+        trust=True,
+    )
+    await ops_test.model.deploy(
+        "prometheus-k8s",
+        application_name=PROMETHEUS_APPLICATION_NAME,
+        trust=True,
+    )
+    await ops_test.model.deploy(
+        "loki-k8s", application_name=LOKI_APPLICATION_NAME, trust=True, channel="stable"
+    )
     await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", timeout=1000)
 
 
+@pytest.mark.abort_on_fail
+async def test_given_tls_access_relation_when_related_and_unrelated_to_notary_then_certificates_replaced_appropriately(
```

Comment:
Where is the `and_unrelated` part of this test?

Reply:
Added back